src/payload.rs

```diff
@@ -260,26 +326,30 @@ impl TransactionDecoder {
     > {
         match payload {
             TransactionPayload::Legacy(legacy) => {
-                let encoding = legacy
-                    .config
-                    .encoding
-                    .unwrap_or(UiTransactionEncoding::Base58);
+                let encoding = match legacy.config.encoding {
+                    Some(enc) => enc,
+                    None => UiTransactionEncoding::Base58,
+                };
+
                 let tx_bytes =
                     TransactionPayload::decode_transaction(&legacy.transaction, encoding)?;
-
                 let tx = bincode::deserialize(&tx_bytes)?;
+
                 Ok((
                     tx,
                     Some(RpcSendTransactionConfigWithBlockList {
                         config: Some(legacy.config),
-                        blocklist_pdas: vec![], // Legacy format doesn't have blocklist
+                        blocklist_pdas: None, // Legacy format doesn't have blocklist
                     }),
                 ))
             }
             TransactionPayload::New(wrapper) => {
                 let tx = bincode::deserialize(&wrapper.transaction)?;
                 let config = if let Some(proto_config) = &wrapper.config {
-                    Some(proto_config.try_into()?)
+                    match proto_config.try_into() {
+                        Ok(config) => Some(config),
+                        Err(err) => return Err(err),
+                    }
                 } else {
                     None
                 };
```

Comment:
```
let config = wrapper
    .config
    .as_ref()
    .map(TryInto::try_into)
    .transpose()?;
```

transpose might help clean this up. Convert `Option<Result<..>>` to `Result<Option<..>>`